src/components/Settings/Settings.react.js

```diff
@@ -679,13 +681,135 @@ class Settings extends Component {
       self.setState(
         {
           uploadingAvatar: false,
+          isAvatarAdded: true,
         },
         self.handleSave(),
       );
     });
   };
 
+  handleAvatarImageChange = e => {
+    e.preventDefault();
+    // eslint-disable-next-line no-undef
+    let reader = new FileReader();
+    let file = e.target.files[0];
+    reader.onloadend = () => {
+      this.setState({
+        file: file,
+        imagePreviewUrl: reader.result,
+        isAvatarAdded: true,
+      });
+    };
+    reader.readAsDataURL(file);
+  };
+
+  removeAvatarImage = () => {
+    this.setState({
+      file: '',
+      isAvatarAdded: false,
+      imagePreviewUrl: '',
+      avatarSrc: '',
+    });
+  };
+
   render() {
+    let avatarRender = null;
```

Comment:
Move this logic to a function named `renderAvatar()` and then return the JSX from that function.